sherpa/models/model.py

```diff
@@ -387,75 +392,95 @@ def modelCacher1d(func):
+        # and the decorator is applied to calc, which is not
+        # called with a integrate kwarg, rather than the call to
+        # somefunc, which was sent an integrate setting.
+        #
+        try:
+            integrate = cls.integrate
+        except AttributeError:
+            # Rely on the integrate kwarg as there's no
+            # model setting.
             #
-            try:
-                integrate = cls.integrate
-            except AttributeError:
-                # Rely on the integrate kwarg as there's no
-                # model setting.
-                #
-                integrate = kwargs.get('integrate', False)
+            integrate = kwargs.get('integrate', False)
 
-            data = [numpy.array(pars).tobytes(),
-                    boolean_to_byte(integrate),
-                    numpy.asarray(xlo).tobytes()]
-            if args:
-                data.append(numpy.asarray(args[0]).tobytes())
+        data = [numpy.array(pars).tobytes(),
+                boolean_to_byte(integrate),
+                numpy.asarray(xlo).tobytes()]
+        if args:
+            data.append(numpy.asarray(args[0]).tobytes())
 
-            token = b''.join(data)
-            digest = hashfunc(token).digest()
-            if digest in cache:
-                cache_ctr['hits'] += 1
-                return cache[digest].copy()
+        # Add any keyword arguments to the list. This will
+        # include the xhi named argument if given. Can the
+        # value field fail here?
+        #
+        for k, v in kwargs.items():
+            data.extend([k.encode(), numpy.asarray(v).tobytes()])
+
+        # Is the value cached?
+        #
+        token = b''.join(data)
+        digest = hashfunc(token).digest()
+        cache = cls._cache
+        if digest in cache:
+            cache_ctr['hits'] += 1
+            return cache[digest].copy()
 
+        # Evaluate the model.
+        #
         vals = func(cls, pars, xlo, *args, **kwargs)
 
-        if use_caching:
-            # remove first item in queue and remove from cache
-            key = queue.pop(0)
-            cache.pop(key, None)
+        # remove first item in queue and remove from cache
+        queue = cls._queue
+        key = queue.pop(0)
+        cache.pop(key, None)
 
-            # append newest model values to queue
-            queue.append(digest)
-            cache[digest] = vals.copy()
+        # append newest model values to queue
+        queue.append(digest)
+        cache[digest] = vals.copy()
 
-            cache_ctr['misses'] += 1
+        cache_ctr['misses'] += 1
 
         return vals
 
     return cache_model
 
+# It is tempting to convert the explicit class names below into calls
+# to super(), but this is problematic since it ends up breaking a
+# number of invariants the classes rely on. An example is that
+# instances of Unary/BinaryOpModel classes should not cache-related
+# attributes, but they can do if we change to using super. There is
```

Comment:
I don't understand this sentence. Do they cache related attributes (relate to what?) or or is this about attributes related to the cache (in that case, there is a verb missing in this sentence)?